src/main/java/com/google/sps/data/SaleCard.java

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import java.time.LocalTime;
+
+/** A saleCard. */
+public final class SaleCard {
+
+  private final long id;
+  private String businessName;
+  private String description;
+  private boolean hasDelivery;
+  private LocalTime startTime;
+  private LocalTime endTime;
+  private LocationData location;
+  private Picture picture;
+
+  public SaleCard(long id, String businessName, String description, boolean hasDelivery,
+      LocalTime startTime, LocalTime endTime, LocationData location, Picture picture) {
+    this.id = id;
+    this.businessName = businessName;
+    this.description = description;
+    this.hasDelivery = hasDelivery;
+    this.startTime = startTime;
+    this.endTime = endTime;
+    this.location = location;
+    this.picture = picture;
+  }
+
+  public SaleCard(EmbeddedEntity embeddedSaleCard) {
```

Comment:
In which case could the `embeddedSaleCard` be null?

Reply:
I added it first thinking of cases when `entity.getProperty("Embedded")` returns null, because the property does not exist or is set as null. I added other checks to handle those cases, so right now there is no case, unless you pass it directly in your code.